Gems/WhiteBox/Code/Source/EditorWhiteBoxComponentMode.cpp

```diff
@@ -303,6 +356,20 @@ namespace WhiteBox
         m_intersectionAndRenderData = {};
         m_currentSubMode = SubMode::Default;
         SetViewportUiClusterActiveButton(m_modeSelectionClusterId, m_defaultModeButtonId);
+
+        // Change sub-mode to default at the next frame to go after the automated mode switching in ComponentModeActionHandler.
+        QTimer::singleShot(
+            0,
```

Comment:
update: I believe it's preferred to use the TickBus next frame call instead of `QTimer::singleShot` (I don't remember the syntax off the top of my head but I know @mbalfour-amzn has suggested this in the past). I think that might be nicer here (and could be easier to test if we want to as well)

Reply:
i did suggest it in the past, though turns out they both have the same flaw. ideally, i'd prefer us to find patterns that didn't require us to do "one tick delays" quite as often in the UI. 

the specific problem i ran into is that if you have any code that brings up a dialog box or anything, Qt will continue ticking inside the dialog box, inside of the main engine tick. The thing I discovered when looking further is that this actually makes the main engine continue ticking reentrantly as well, which is pretty awkward. So either way, you can have these "one tick delays" actually fire _sort of_ in the same tick that they're created in with the right circumstances.

ideally, we'd want code like this to be more deterministically hooked up to the exact event or circumstance that they need to have completed before running the next stage of logic. so for example, if it needs to run after "automated mode switching", it would queue an action to run on an automated mode switch, not on a tick.

Reply:
`QueueFunction` I think was the name, but @mbalfour-amzn does make a good point above

Reply:
I agree that this should not be necessary. If this same issue pops up in another one of the migrations, I'll try to refactor this so that the timeout isn't necessary.